source/io.c

```diff
@@ -98,14 +98,12 @@ static void do_misaligned_buffer_read(
     const uacpi_buffer_field *field, uacpi_u8 *dst
 )
 {
-    struct bit_span src_span = {
-        .index = field->bit_index,
-        .length = field->bit_length,
-        .const_data = field->backing->data,
-    };
-    struct bit_span dst_span = {
-        .data = dst,
-    };
+    struct bit_span src_span = {0};
+    struct bit_span dst_span = {0};
+    src_span.index = field->bit_index;
+    src_span.length = field->bit_length;
+    src_span.const_data = field->backing->data;
+    dst_span.data = dst;
```

Comment:
maybe move the `dst_span` init code next to the `dst_span.length` assignment below and leave an empty line here to logically separate them?